File: app/src/main/java/com/synapse/social/studioasinc/backend/IDatabaseService.java
package com.synapse.social.studioasinc.backend;

import com.synapse.social.studioasinc.backend.interfaces.IDataListener;
import com.synapse.social.studioasinc.backend.interfaces.ICompletionListener;
import java.io.File;
import java.util.Map;

public interface IDatabaseService {
    IDatabaseReference getReference(String path);
    void getUserById(String uid, IDataListener listener);
    void setValue(String path, Object value, ICompletionListener listener);
    void updateChildren(String path, Map<String, Object> children, ICompletionListener listener);
    void uploadFile(File file, String path, ICompletionListener listener);
    void searchUsers(String query, IDataListener listener);
    void getFollowers(String uid, IDataListener listener);
    void getFollowing(String uid, IDataListener listener);
    void getData(String path, IDataListener listener);
}

File: app/src/main/java/com/synapse/social/studioasinc/backend/IDatabaseService.kt
package com.synapse.social.studioasinc.backend

import com.synapse.social.studioasinc.backend.interfaces.ICompletionListener
import com.synapse.social.studioasinc.backend.interfaces.IDataListener
import java.io.File

interface IDatabaseService {
    fun getReference(path: String): IDatabaseReference
    fun getUserById(uid: String, listener: IDataListener)
    fun setValue(path: String, value: Any, listener: ICompletionListener<*>)
    fun updateChildren(path: String, children: MutableMap<String, Any>, listener: ICompletionListener<*>)
    fun uploadFile(file: File, path: String, listener: ICompletionListener<*>)
    fun searchUsers(query: String, listener: IDataListener)
    fun getFollowers(uid: String, listener: IDataListener)
    fun getFollowing(uid: String, listener: IDataListener)
    fun getData(path: String, listener: IDataListener)
}